models/DataObject/ClassDefinition/Data.php

```diff
@@ -833,7 +840,14 @@ public function getSetterCodeObjectbrick($brickClass)
         }
 
         if ($this->supportsDirtyDetection()) {
+            $code .= "\t" . 'if(\Pimcore\Model\DataObject::doGetInheritedValues($this->getObject())) {' . "\n";
```

Comment:
This looks wrong to me and probably causes the tests to fail.
I think it should be something like 

```
        if ($this->supportsDirtyDetection()) {
            $code .= "\t" . '$class = $this->getObject() ? $this->getObject()->getClass() : null;' . "\n";
            $code .= "\t" . 'if($class && $class->getAllowInherit()) {' . "\n";
            $code .= "\t\t" . '$inheritValues = $this->getObject()::getGetInheritedValues();'."\n";
            $code .= "\t\t" . '$this->getObject()::setGetInheritedValues(false);'."\n";
            $code .= "\t" . '}'."\n";
            $code .= "\t" . '$currentData = $this->get' . ucfirst($this->getName()) . '();' . "\n";
            $code .= "\t" . 'if($class && $class->getAllowInherit()) {' . "\n";
            $code .= "\t\t" . '$this->getObject()::setGetInheritedValues($inheritValues);'."\n";
            $code .= "\t" . '}' . "\n";
            $code .= "\t" . '$isEqual = $fd->isEqual($currentData, $' . $key . ');' . "\n";
            $code .= "\t" . 'if (!$isEqual) {' . "\n";
            $code .= "\t\t" . '$this->markFieldDirty("' . $key . '", true);' . "\n";
            $code .= "\t" . '}' . "\n";
        }
```

Reply:
You are right. Committed those changes - let's look what Travis thinks about it ;-)